mifosng-android/src/main/java/com/mifos/mifosxdroid/online/loanrepaymentschedule/LoanRepaymentScheduleScreen.kt

```diff
@@ -0,0 +1,359 @@
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.tooling.preview.PreviewParameterProvider
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import com.mifos.core.designsystem.component.MifosCircularProgress
+import com.mifos.core.designsystem.component.MifosScaffold
+import com.mifos.core.designsystem.component.MifosSweetError
+import com.mifos.core.designsystem.icon.MifosIcons
+import com.mifos.core.objects.accounts.loan.LoanWithAssociations
+import com.mifos.core.objects.accounts.loan.Period
+import com.mifos.core.objects.accounts.loan.RepaymentSchedule
+import com.mifos.mifosxdroid.R
+import com.mifos.utils.DateHelper
+
+/**
+ * Created by Pronay Sarker on 03/07/2024 (9:18 AM)
+ */
+
+@Composable
+fun LoanRepaymentScheduleScreen(
+    loanId: Int,
+    viewModel: LoanRepaymentScheduleViewModel = hiltViewModel(),
+    navigateBack: () -> Unit
+) {
+    val uiState by viewModel.loanRepaymentScheduleUiState.collectAsStateWithLifecycle()
+
+    LaunchedEffect(key1 = Unit) {
+        viewModel.loadLoanRepaySchedule(loanId)
+    }
+
+    LoanRepaymentScheduleScreen(
+        uiState = uiState,
+        navigateBack = navigateBack,
+        onRetry = { viewModel.loadLoanRepaySchedule(loanId) }
+    )
+}
+
+@Composable
+fun LoanRepaymentScheduleScreen(
+    uiState: LoanRepaymentScheduleUiState,
+    navigateBack: () -> Unit,
+    onRetry: () -> Unit
+) {
+    val snackbarHostState = remember {
+        SnackbarHostState()
+    }
+
+    MifosScaffold(
+        title = stringResource(R.string.loan_repayment_schedule),
+        snackbarHostState = snackbarHostState,
+        icon = MifosIcons.arrowBack,
+        onBackPressed = navigateBack
+    ) {
+        Box(modifier = Modifier.padding(it)) {
+            when (uiState) {
+                is LoanRepaymentScheduleUiState.ShowFetchingError -> {
+                    MifosSweetError(
+                        message = uiState.message,
+                        onclick = onRetry
+                    )
+                }
+
+                is LoanRepaymentScheduleUiState.ShowLoanRepaySchedule -> {
+                    LoanRepaymentScheduleContent(
+                        uiState.loanWithAssociations.repaymentSchedule.getlistOfActualPeriods()
+                    )
+                }
+
+                LoanRepaymentScheduleUiState.ShowProgressbar -> {
+                    MifosCircularProgress()
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun LoanRepaymentScheduleContent(
+    periods: List<Period>
+) {
+    Box(
+        modifier = Modifier.fillMaxSize()
+    ) {
+        Column(
+            modifier = Modifier.fillMaxSize()
+        ) {
+            HeaderLoanRepaymentSchedule()
+
+            LazyColumn {
```

Comment:
Above Box and Column does not make sense because they are just wrapping, we can remove and set `modifier = Modifier.fillMaxSize()` here in LazyColumn.

Reply:
I did not saw BottomBarLoanRepaymentSchedule, you can achieve this by using Box and Lazy Column, 

Put LazyClumn inside Box (Box is similar to FrameLayout so you can have fragment front and back)